texar/core/optimization.py

```diff
@@ -271,12 +272,17 @@ def get_grad_clip_fn(hparams: Optional[Union[HParams,
     return grad_clip_fn
 
 
-def get_train_op(optimizer: Optimizer,
+def get_train_op(params: Optional[Union[List[torch.Tensor],
+                                        List[Dict[str,
+                                                  List[torch.Tensor]]]]] = None,
+                 optimizer: Optional[Optimizer] = None,
```

Comment:
`get_train_op` should also have an argument `scheduler=None` to allow users to pass in any customized scheduler, like optimizer.